src/VisualStudio/TestUtilities/IntegrationHost.cs

```diff
@@ -0,0 +1,522 @@
+                {
+                    throw new Exception($"The specified registry key could not be found. Registry Key: '{registryKey}'");
+                }
+
+                return Path.GetFullPath(registryKey.GetValue("InstallDir").ToString());
+            }
+        }
+
+        private static string LocateVsUserFilesFolder()
+        {
+            using (var registryKey = Registry.CurrentUser.OpenSubKey(Path.Combine(VsConfigRootSubKey, "Initialization"), false))
+            {
+                if (registryKey == null)
+                {
+                    throw new Exception($"The specified registry key could not be found. Registry Key: '{registryKey}'");
+                }
+
+                return Path.GetFullPath(registryKey.GetValue("UserFilesFolder").ToString());
+            }
+        }
+
+        private static DTE LocateDteForProcess(Process process)
+        {
+            var dte = null as object;
+            var runningObjectTable = null as IRunningObjectTable;
+            var enumMoniker = null as IEnumMoniker;
+            var bindContext = null as IBindCtx;
+            var monikers = new IMoniker[1];
+
+            Ole32.GetRunningObjectTable(0, out runningObjectTable);
+            runningObjectTable.EnumRunning(out enumMoniker);
+            Ole32.CreateBindCtx(0, out bindContext);
+
+            do
+            {
+                monikers[0] = null;
+
+                var monikersFetched = 0u;
+                var result = enumMoniker.Next(1, monikers, out monikersFetched);
+
+                if (result != 0)
+                {
+                    return null;
+                }
+
+                var moniker = monikers[0];
+                var fullDisplayName = null as string;
+
+                moniker.GetDisplayName(bindContext, null, out fullDisplayName);
+
+                var displayNameProcessId = 0;
+
+                if (!int.TryParse(fullDisplayName.Split(':').Last(), out displayNameProcessId))
+                {
+                    continue;
+                }
+
+                var displayName = fullDisplayName.Substring(0, (fullDisplayName.Length - (displayNameProcessId.ToString().Length + 1)));
+                var fullProgId = VsProgId.StartsWith("!") ? VsProgId : $"!{VsProgId}";
+
+                if (displayName.Equals(fullProgId, StringComparison.OrdinalIgnoreCase))
+                {
+                    if (displayNameProcessId == process.Id)
+                    {
+
+                        runningObjectTable.GetObject(moniker, out dte);
+                    }
+                }
+            }
+            while (dte == null);
+
+            return (DTE)(dte);
+        }
+
+        public void Cleanup()
+        {
+            // Clean the Backup Files folder so we avoid the recovery dialog on launch.
+            IntegrationHelper.DeleteDirectoryRecursively(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), VsUserFilesFolder, "Backup Files"));
```

Comment:
It looks like this design means this type of test can never be run in parallel on the same machine.  Is that the case?  

Fine if the answer is yes but I want to make sure that is an explicit part of the design.